modules/nw-operator-vsphere-multiple-nic.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_vsphere/ipi/installing-vsphere-installer-provisioned-network-customizations.adoc
+// * installing/installing_vsphere/upi/installing-vsphere-network-customizations.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-operator-vsphere-multiple-nic_{context}"]
+= Specifying multiple network interface controllers (NICs)
+
+Before you install an {product-title} cluster on a {vmw-short} host, you can specify multiple network interface controllers (NICs) for a networking implementation so that the {vmw-short} cloud controller manager (CCM) can select the appropriate NIC for a given networking situation. The {vmw-full} can use the NIC for managing pods and services on your cluster. For this configuration, you must specify internal and external Classless Inter-Domain Routing (CIDR) implementations in the {vmw-short} cloud controller manager (CCM) configuration. Each CIDR implementation lists an IP address range that the CCM uses to determine what NICs interact with traffic from internal and external networks.
+
+If you do not configure internal and external CIDR implementations in the {vmw-short} CCM configuration, the {vmw-short} CCM selects the first NIC that the host operating system reports as the primary network interface. This configuration can cause new nodes that associate with a `MachineSet` object with a single NIC to become unusable as each new node receives the `node.cloudprovider.kubernetes.io/uninitialized` taint. In certain situations, the reported network interface might not be the expected network interface. These situations can cause communication issues with the Kubernetes API server that can cause installation of the cluster to fail.
+
+[IMPORTANT]
+====
+If you configured multiple NICs for a network implementation but intend to use only a single NIC for your current needs, ensure that you specify the same IP address for the internal and external CIDR implementations. This configuration ensures that the CCM selects your preferred NIC and not the first NIC that the host operating system reports.
```

Comment:
Thanks. `virt` platforms do not support multiple NICs in general? Updated to state:

> If you configured multiple NICs for a network implementation but you intend to use only a single NIC, ensure that you deploy your cluster nodes on the single NIC. For a single NIC configuration, you must specify the same IP address for the internal and external CIDR implementations. This configuration ensures that the CCM selects your preferred NIC and not the first NIC that the host operating system reports.

> Virtualization platforms, such as {vmw-first} ESXi, do not support configuring a multiple NIC environment.

Reply:
No, multiple nics are fine, that's the reason we need to document this configuration. People shouldn't be deploying with unused nics though. There's no reason to populate your VMs with two nics and then only use one.

Also, keep in mind that this documentation is VSphere-specific. We don't necessarily need to specify in each section that it relates to VSphere.

Reply:
Thanks. How about:

```
To prevent any NIC selection issues, only configure the amount of NICs that you will use. If you configured multiple NICs for a network implementation but you intend to use only a single NIC, ensure that you deploy your cluster nodes on the single NIC.

For a single NIC configuration, you must specify the same IP address for the internal and external CIDR implementations. This configuration ensures that the CCM selects your preferred NIC and not the first NIC that the host operating system reports.
```